components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -432,63 +442,300 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
+
+    /**
+     * Get all attributes for given domain.
+     *
+     * @param domainName
+     * @return
+     * @throws UserStoreException
+     */
+    private Map<String, String> getAllAttributes(String domainName) throws UserStoreException {
+
+        ClaimMapping[] userClaims;
+        ClaimMapping[] coreClaims;
+        ClaimMapping[] extensionClaims = null;
+
+        try {
+            coreClaims = carbonClaimManager.getAllClaimMappings(SCIMCommonConstants.SCIM_CORE_CLAIM_DIALECT);
+            userClaims = carbonClaimManager.getAllClaimMappings(SCIMCommonConstants.SCIM_USER_CLAIM_DIALECT);
+            if (SCIMUserSchemaExtensionBuilder.getInstance().getExtensionSchema() != null) {
+                extensionClaims = carbonClaimManager.getAllClaimMappings(
+                        SCIMUserSchemaExtensionBuilder.getInstance().getExtensionSchema().getURI());
+            }
+            Map<String, String> attributes = new HashMap<>();
+            for (ClaimMapping claim : coreClaims) {
+                attributes.put(claim.getClaim().getClaimUri(), claim.getMappedAttribute(domainName));
+            }
+            for (ClaimMapping claim : userClaims) {
+                attributes.put(claim.getClaim().getClaimUri(), claim.getMappedAttribute(domainName));
+            }
+            if (extensionClaims != null) {
+                for (ClaimMapping claim : extensionClaims) {
+                    attributes.put(claim.getClaim().getClaimUri(), claim.getMappedAttribute(domainName));
+                }
+            }
+            return attributes;
+        } catch (UserStoreException e) {
+            throw new UserStoreException("Error in filtering users by multi attributes ", e);
+        }
+    }
 
+    /**
+     * Perform multi attribute filtering.
+     *
+     * @param node
+     * @param offset
+     * @param limit
+     * @param sortBy
+     * @param sortOrder
+     * @param domainName
+     * @return
+     * @throws CharonException
+     */
+    private String[] getFilteredUsersFromMultiAttributeFiltering(Node node, int offset, int limit, String sortBy,
+                                                                 String sortOrder, String domainName)
+            throws CharonException {
+
+        String[] userNames;
+
+        try {
+            if (StringUtils.isEmpty(domainName)) {
+                domainName = "PRIMARY";
+            }
+            Map<String, String> attributes = getAllAttributes(domainName);
+            if (log.isDebugEnabled()) {
+                log.debug("Invoking the do get user list for domain: " + domainName);
+            }
+            userNames = ((PaginatedUserStoreManager) carbonUM).getUserList(getCondition(node, attributes), domainName,
+                    UserCoreConstants.DEFAULT_PROFILE, limit, offset, sortBy, sortOrder);
+            return userNames;
+        } catch (UserStoreException e) {
+            throw new CharonException("Error in filtering users by multi attributes ", e);
```

Comment:
lets add more context to the error message.

Reply:
Added